File: Backend/Models/closeoutForm.js
import mongoose from 'mongoose';

const closeoutFormSchema = new mongoose.Schema({
    sreeningId: {
        type: String,
        unique: true,
        required: true
    },
    closeOutInterviewDate: {
        type: Date,
        required: true,
        default: Date.now
    },
    dateOfTermination: {
        type: Date,
        required: true,
        default: Date.now
    },
    participantStatus: {
        choicesStudy: {
            type: String,
            enum: ["Completed study visits", "Participation terminated prior to completion of study visits", "Screen failure before enrollment"],
            required: true
        },
        incompleteReason: {
            incompletionOptions: {
                type: String,
                enum: ["Adverse event", "Death", "Lost to follow-up", "Physician decision", "Protocol deviation", "Screen failure", "Study terminated by sponsor", "Withrawal by participant", "Other"],
                required: false
            },
            adverseEvent: { type: String, required: false },
            deathOption: { type: Date, required: false },
            protocalDeviation: { type: String, required: false },
            withdrawalReason: { type: String, required: false },
            otherReason: { type: String, required: false },
        }
    },
    submittedBy: { type: String, required: false },
    submittedAt: { type: Date, default: Date.now }
});

const closeoutForm = mongoose.model('closeoutForm', closeoutFormSchema);

export default closeoutForm;

File: api/Models/CloseoutForm.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace KemriApi.Models
{
    public class closeoutForm
    {
        [BsonId]
        [BsonRepresentation(BsonType.ObjectId)]
        public string? Id { get; set; }

        [BsonElement("sreeningId")]
        public string screeningId { get; set; } = string.Empty;

        [BsonElement("closeOutInterviewDate")]
        public DateTime closeOutInterviewDate { get; set; } = DateTime.UtcNow;

        [BsonElement("dateOfTermination")]
        public DateTime dateOfTermination { get; set; } = DateTime.UtcNow;

        [BsonElement("participantStatus")]
        public ParticipantStatusInfo participantStatus { get; set; } = new();

        [BsonElement("submittedBy")]
        public string? submittedBy { get; set; }

        [BsonElement("submittedAt")]
        public DateTime submittedAt { get; set; } = DateTime.UtcNow;
    }

    public class ParticipantStatusInfo
    {
        [BsonElement("choicesStudy")]
        public string choicesStudy { get; set; } = string.Empty;

        [BsonElement("incompleteReason")]
        public IncompleteReasonInfo? incompleteReason { get; set; }
    }

    public class IncompleteReasonInfo
    {
        [BsonElement("incompletionOptions")]
        public string? incompletionOptions { get; set; }

        [BsonElement("adverseEvent")]
        public string? adverseEvent { get; set; }

        [BsonElement("deathOption")]
        public DateTime? deathOption { get; set; }

        [BsonElement("protocalDeviation")]
        public string? protocalDeviation { get; set; }

        [BsonElement("withdrawalReason")]
        public string? withdrawalReason { get; set; }

        [BsonElement("otherReason")]
        public string? otherReason { get; set; }
    }
}
